src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -108,15 +110,39 @@ public void setRestoreToDemoUser() {
         mRestoreStrategySet = true;
     }
 
-    public void run() {
+    public void advanceSessionWithEndpoint(String[] endpointIdAndArgs) {

```

Comment:
In the other platforms, I believe we explicitly reset the session state before we begin any endpoint execution. Even if the CLI can't currently end up in that state it seems worth being consistent with the execution structure.

Reply:
Makes sense. Does resetting the state here mean `clearVolatiles`, or something else?

Reply:
I think this is the android entrypoint

https://github.com/dimagi/commcare-android/blob/35627e56cbe693cf8951e0315ca2f33057b3fc5c/app/src/org/commcare/activities/HomeScreenBaseActivity.java#L222

which follows to 
https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/models/AndroidSessionWrapper.java#L70

so I think those two steps are the right ones.

Reply:
a777f9aa4